src/macros.jl

```diff
@@ -543,6 +556,18 @@ function build_constraint(
     return VectorConstraint(x, MOI.SOS2{Float64}(set.weights))
 end
 
+function build_non_relational_constraint(_error::Function, head, args...) # This would remain in JuMP.
```

Comment:
Why not dispatch on the `head` in `parse_...` as done in `parse_one_operator_constraint` ?
`build_constraint` should have 3 arguments: `_error`, `func`, `set` that builds the constraint `func`-in-`set`.
`parse_..._constraint` is where the function and sets are defined.